src/services/worker/tasks/burn-rates.task.ts

```diff
@@ -34,53 +34,58 @@ export class BurnRatesTask {
   })
   async syncBurnRates(): Promise<void> {
-    const missedRecords = await this._addressService.findBy({
-      where: { burnRate: null },
-      select: ['ownerAddress'],
+    const missedRecords = (
+      await this._clusterService.findBy({
+        where: { burnRate: null },
+      })
+    ).map(item => {
+      item.cluster = JSON.parse(item.cluster);
+      return item;
     });
     const burnRates = await Promise.allSettled(
-      missedRecords.map(({ ownerAddress }) =>
-        Web3Provider.burnRate(ownerAddress),
-      ),
+      missedRecords.map(({ owner, operatorIds, cluster }) => {
+        return Web3Provider.getClusterBurnRate(owner, operatorIds, cluster);
+      }),
     );
     const balances = await Promise.allSettled(
-      missedRecords.map(({ ownerAddress }) =>
-        Web3Provider.totalBalanceOf(ownerAddress),
+      missedRecords.map(({ owner, operatorIds, cluster }) =>
+        Web3Provider.getBalance(owner, operatorIds, cluster),
       ),
     );
     const liquidated = await Promise.allSettled(
-      missedRecords.map(({ ownerAddress }) =>
-        Web3Provider.isLiquidated(ownerAddress),
+      missedRecords.map(({ owner, operatorIds, cluster }) =>
+        Web3Provider.isLiquidated(owner, operatorIds, cluster),
       ),
     );
     const currentBlockNumber = await Web3Provider.currentBlockNumber();
-    const liquidationThresholdPeriod =
-      await Web3Provider.liquidationThresholdPeriod();
+    const minimumBlocksBeforeLiquidation =
+      await Web3Provider.minimumBlocksBeforeLiquidation();
 
     for (const [index, record] of missedRecords.entries()) {
       const burnRate = +(burnRates[index] as any).value;
       const balanceObject = balances[index] as any;
       const balance =
         balanceObject.status === 'rejected' &&
-        balanceObject.reason.toString().includes('negative balance')
+        balanceObject.reason.toString().includes('InsufficientFunds')
           ? 0
           : +balanceObject.value;
       const isLiquidated = (liquidated[index] as any).value;
       record.burnRate = burnRate;
-      record.liquidateLastBlock =
+      record.cluster = JSON.stringify(record.cluster);
+      record.balanceToBlockNumber =
         burnRate > 0
           ? currentBlockNumber + +(balance / burnRate).toFixed(0)
           : null;
-      record.liquidateFirstBlock = record.liquidateLastBlock
-        ? record.liquidateLastBlock - liquidationThresholdPeriod
+      record.liquidationBlockNumber = record.balanceToBlockNumber
+        ? record.balanceToBlockNumber - minimumBlocksBeforeLiquidation
         : null;
       record.balance = balance;
       record.isLiquidated = isLiquidated;
     }
     await Promise.all(
       missedRecords.map(record =>
-        this._addressService.update(
-          { ownerAddress: record.ownerAddress },
+        this._clusterService.update(
+          { owner: record.owner, operatorIds: record.operatorIds },
           record,
         ),
       ),
```

Comment:
In `syncBurnRates` we definitely need comments explaining common parts of algorithm